internal/webhook/managedcluster_webhook.go

```diff
@@ -298,3 +336,38 @@ func getMissingProvidersWithWrongVersions(exposed, required []hmcv1alpha1.Provid
 
 	return missing, nonSatisfying, merr
 }
+
+func isCredMatchTemplate(cred *hmcv1alpha1.Credential, template *hmcv1alpha1.ClusterTemplate) error {
+	idtyKind := cred.Spec.IdentityRef.Kind
+	infraProviders := template.Status.Providers.InfrastructureProviders
+
+	errMsg := func(idtyKind string, provider string) error {
+		return fmt.Errorf("wrong kind of the ClusterIdentity \"%s\" for provider \"%s\"", idtyKind, provider)
+	}
+
+	if len(infraProviders) == 0 {
```

Comment:
wdyt about checking providers' absence before making the API calls (getting the `credentials` object)? i think it might save some network in theory

Reply:
if there are no providers available in a template, then we will fetch credentials (assuming it's ready) each time just to declare that there are no providers available, to me it looks a bit redundant, so it actually does not matter whether the creds are ready or not — the validation will fail either way

Reply:
tbh I still don't see much point in this, but I will fix it

Reply:
thanks. the main point is to avoid making unnecessary network invocations, simply this :)